pom.xml

```diff
@@ -279,6 +279,9 @@
 			<plugin>
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
+		                <configuration>
```

Comment:
Additionally, we need to get more clarity on the discussion of #197. 
It also means that we need to solve the [matter of "public solution model"](https://github.com/promregator/promregator/discussions/197#discussioncomment-743876). Perhaps adding some annotations to indicate whether some API is public or not would also be sufficient?

Reply:
This is another great point in terms of maintenance and long-term flexibility. Right now, it is not immediately clear what is available to be extended / overridden safely. Clearly, this is part of a larger discussion as we need to determine what we think is going to remain a constant, stable interface and what is actually implementation details subject to change with new versions of CloudFoundry, etc., etc.

My personal experience is that we'd want to make very clear "why" (what reasons) a user would generally want to extend Promregator (i.e. custom credentials implementation, custom verification and caching of those creds, etc.).  Things outside of that list would be "at your own risk" in terms of extension - unless of course, the user opened a PR for review here.

Reference to your mention here: https://github.com/promregator/promregator/discussions/197#discussioncomment-743889, we'd need to consider the same issue for any additional library introduced.

We can absolutely use annotations as suggested or whatever solution makes the most sense. Another (higher effort) option would be package structures which make it clear what is going to remain backward compatible and what is internal.